src/Client.java

```diff
@@ -1,48 +1,23 @@
-import java.io.IOException;
-import java.net.*;
-
 public class Client {
     private final int PORT = 9876;
     private final String SERVER = "localhost";
-    private DatagramSocket clientSocket;
-    private byte[] buf;
+    private Protocol protocol;
 
-    public Client() {
-        try {
-            clientSocket = new DatagramSocket();
-        } catch (SocketException e) {
-            e.printStackTrace();
+    public Client(String type) {
+        if(type.equals("udp")){
+            protocol = new UDPProtocol();
```

Comment:
Should the variable name be protocolSocket?

Reply:
I think the protocol would be better because the protocol does send and receive a packet.